h2o-core/src/main/java/water/H2O.java

```diff
@@ -2048,6 +2059,9 @@ public static void main( String[] args ) {
     // Log registered parsers
     Log.info("Registered parsers: " + Arrays.toString(ParserService.INSTANCE.getAllProviderNames(true)));
 
+    // Start thread checking client disconnections
+    new ClientDisconnectCheckThread().start();
```

Comment:
Is this thread started even for the client itself?

Reply:
yes, it is started for each node. It seems easier to handle each node the same than handling client and non-clients nodes differently. What do you think?